stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -269,6 +269,78 @@ function do_string_escape(s)
     return escape_string(s, ('\"','$'))
 end
 
+const PATH_cache_lock = Base.ReentrantLock()
+const PATH_cache = Set{String}()
+cached_PATH_string::Union{String,Nothing} = nothing
+function cached_PATH_changed()
+    global cached_PATH_string
+    @lock(PATH_cache_lock, cached_PATH_string) !== get(ENV, "PATH", nothing)
+end
+const PATH_cache_finished = Base.Condition() # used for sync in tests
```

Comment:
`Base.Condition` is deprecated, since it is only usable in single-threaded programs. Use an Event or `Threads.Condition` instead

Reply:
Yeah https://github.com/JuliaLang/julia/pull/52893 fixes this